### PR TITLE
Added warning message when street has no house number in cat17

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
https://phabricator.wikimedia.org/T180042

Based on code from 10h16

### CODE REVIEW

File: skins/cat17/src/sass/common/_variables.scss [hunk 1]
```diff
@@ -20,6 +20,8 @@ $brand-footer: #cfdbe2;
 $gray-light: #ddd;
 $brand-error: #812923;
 $brand-error-rgba: rgba(129, 41, 35, 0.2);
+$brand-warning: #812923;
+$brand-warning-rgba: rgba(129, 41, 35, 0.2);
```
Comment (wiese) [inline skins/cat17/src/sass/common/_variables.scss]: Seems to be unused.

File: skins/cat17/src/scripts/donationForm.js [hunk 2]
```diff
@@ -295,7 +295,16 @@ $( function () {
 				'donationFormContent.email',
 				WMDE.StateAggregation.donorTypeAndAddressAreValid
 			]
-		}
+		},
+      {
+        viewHandler: WMDE.View.createWarningBoxHandler(
```
Comment (wiese) [inline skins/cat17/src/scripts/donationForm.js]: We probably can get away without a new view handler (warning_box.js)

      {
	  	viewHandler: WMDE.View.createElementClassSwitcher(
			$( '#street, #adress-company' ).nextAll( '.warning-text' ),
			/^([^0-9]*)$/,
			'problematic'
		),
		stateKey: 'donationFormContent.street'
      }

File: skins/cat17/src/sass/common/_variables.scss (hunk 1 above)
Comment (JeroenDeDauw) [reply skins/cat17/src/sass/common/_variables.scss]: Do you mean just the rgba one? I though that for whatever reason both these are needed. Not?

Comment (wiese) [reply skins/cat17/src/sass/common/_variables.scss]: Yes, I just refered to the -rgba one. There is no magic involved, you just have to keep `skins/cat17/node_modules/bootstrap-sass/assets/stylesheets/bootstrap/_variables.scss` in mind. You got lucky, used the "correct" (pre-existing) variable `brand-warning`.

File: skins/cat17/src/scripts/donationForm.js (hunk 2 above)
Comment (JeroenDeDauw) [reply skins/cat17/src/scripts/donationForm.js]: Ah yes. Was that stuff added after error_box or something? I've now removed warning_box and used `createSimpleVisibilitySwitcher` instead.

File: skins/cat17/src/sass/common/_variables.scss (hunk 1 above)
Comment (JeroenDeDauw) [reply skins/cat17/src/sass/common/_variables.scss]: Killed the line

**Pull request merged**